test/model/message_test.dart

```diff
@@ -1733,13 +1780,25 @@ void main() {
 
       test('other flags not affected', () async {
         final message = eg.streamMessage(flags: [MessageFlag.starred, MessageFlag.read]);
-        await prepare();
+        await prepare(starredMessages: [message.id]);
         await prepareMessages([message]);
         await store.handleEvent(mkRemoveEvent(MessageFlag.read, [message]));
         checkNotifiedOnce();
         check(store).messages.values
           .single.flags.deepEquals([MessageFlag.starred]);
       });
+
+      test('remove from starredMessages', () async {
+        final message1 = eg.streamMessage(flags: [MessageFlag.starred]);
+        final message2 = eg.streamMessage(flags: [MessageFlag.starred]);
+        await prepare(starredMessages: [message1.id, message2.id]);
+        await prepareMessages([message2]);
+        check(store).starredMessages.deepEquals([message1.id, message2.id]);
+        await store.handleEvent(
+          mkRemoveEvent(MessageFlag.starred, [message1, message2]));
+        checkPerAccountStoreNotifiedOnce();
```

Comment:
These checks aren't totally satisfying because they don't verify whether the MessageListView notified its listeners or not.

I'll push an added commit on top which reorganizes these checks a bit and does that.